warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelper.java

```diff
@@ -0,0 +1,242 @@
+     * </ul>
+     * If a match is found, a copy of the source field's NormalizedContentInterface objects is created and added to the whindex fields under the configured
+     * destination field.
+     * </p>
+     *
+     * @param eventMap
+     *            the input event map containing field names and their corresponding normalized content.
+     * @return a multimap of whindex fields generated from the event map.
+     */
+    public Multimap<String,NormalizedContentInterface> processWhindexFields(Multimap<String,NormalizedContentInterface> eventMap) {
+        Multimap<String,NormalizedContentInterface> whindicesInEventMap = HashMultimap.create();
+
+        // Filter whindex rules for which both the value field and source field are present in the event map.
+        List<WhindexConfig> matchingConfigs = valueFieldsToWhindexConfigs.entries().stream()
+                        .filter(entry -> eventMap.containsKey(entry.getValue().getValueField()) && eventMap.containsKey(entry.getValue().getSourceField()))
+                        .map(Map.Entry::getValue).collect(Collectors.toList());
+
+        // Iterate over the matching configurations to generate the whindex fields.
+        for (WhindexConfig currWhindexConfig : matchingConfigs) {
+            // Retrieve all normalized content corresponding to the value field.
+            Collection<NormalizedContentInterface> relatedValueEventContents = eventMap.get(currWhindexConfig.getValueField());
+            // Check if any of these contents match the allowed values (either event field or indexed field).
+            boolean containsAnyMatchingValue = relatedValueEventContents.stream().anyMatch(nci -> {
+                List<String> currentValues = currWhindexConfig.getValues();
+                return currentValues.contains(nci.getEventFieldValue()) || currentValues.contains(nci.getIndexedFieldValue());
+            });
+
+            if (containsAnyMatchingValue) {
+                // Retrieve all normalized content corresponding to the source field.
+                Collection<NormalizedContentInterface> relatedSourceEventContents = eventMap.get(currWhindexConfig.getSourceField());
+                List<NormalizedContentInterface> copies = new ArrayList<>();
+                // Create copies of the source field's normalized content.
+                for (NormalizedContentInterface content : relatedSourceEventContents) {
+                    NormalizedFieldAndValue copy = new NormalizedFieldAndValue(content);
+                    copies.add(copy);
+                }
+                // Map the copies to the destination field (whindex field).
+                whindicesInEventMap.putAll(currWhindexConfig.getDestField(), copies);
+
+                // Removes the entry from the eventMap IFF a whindex field is created and the srcField is overloaded.
+                if (currWhindexConfig.isOverloaded()) {
+                    eventMap.removeAll(currWhindexConfig.getSourceField());
```

Comment:
You should wait until you've iterated through each whindex config before deleting any overloaded source fields. It is possible for multiple whindex configs to reference the same source field, and deleting the source fields too early can lead to missing whindex entries. Something like the following should work:

```java
Set<String> fieldsToRemove = new HashSet<>();
for(WhindexConfig currWhindexConfig : matchingConfigs) {
    // Create whindex entries. Add overloaded src fields to fieldsToRemove when a corresponding whindex entry is added.
}

for (String field : fieldsToRemove) {
    // Now delete the src fields.
}
```

Reply:
This would also be a good edge case to add a test for: if multiple configs with the same overloaded src fields are called, do we still get all the expected whindex entries?